pkg/deployment/collector.go

```diff
@@ -64,8 +64,15 @@ func (c *Collector) Get() *appsv1.Deployment {
 		})
 	}
 
+	storageType := c.jaeger.Spec.Storage.Type
+	// If ingester options have been defined, then change storage type
+	// to Kafka, and the storage options will be used in the Ingester instead
+	if len(c.jaeger.Spec.Ingester.Options.Map()) > 0 {
```

Comment:
This is interesting, do we assume ingester options have to be provided? 

Or in other words are not we supporting the case where no conf props are needed?

Reply:
This was the easiest approach, as I believe ingester options will always be required - but is not intuitive. As mentioned in the PR description, I think a better approach would be to have more distinct strategies that effectively define the architectural structure being used. So all-in-one, another for the current production setup, and a new third option used to represent the kafka based approach.